conda_lock/pypi_solver.py

```diff
@@ -140,6 +129,110 @@ def get_marker_env(self) -> Dict[str, str]:
         }
 
 
+def _extract_glibc_version_from_virtual_packages(
+    platform_virtual_packages: Dict[str, dict]
+) -> Optional[Version]:
+    """Get the glibc version from the "package" repodata of a chosen platform.
+
+    >>> platform_virtual_packages = {
+    ...     "__glibc-2.17-0.tar.bz2": {
+    ...         "name": "__glibc",
+    ...         "version": "2.17",
+    ...     },
+    ... }
+    >>> _extract_glibc_version_from_virtual_packages(platform_virtual_packages)
+    <Version('2.17')>
+    >>> _extract_glibc_version_from_virtual_packages({}) is None
+    True
+    """
+    matches: List[Version] = []
+    for p in platform_virtual_packages.values():
+        if p["name"] == "__glibc":
+            matches.append(Version(p["version"]))
+    if len(matches) == 0:
+        return None
+    elif len(matches) == 1:
+        return matches[0]
+    else:
+        lowest = min(matches)
```

Comment:
this will technically not work for those "special cases" because Version("2010") > Version("2.27") I suspect. I don't think this is a case that is very common though so maybe we can ignore it or error out? Or special case the special cases like you do below.

Reply:
According to my understanding, there is no glibc v2010, so this should never happen. I made an attempt to explain this in the docstring.

Reply:
Ah yes, you are right, 2010 was actually the old manylinux name for 2.12 so yes, the glibc version would be 2.12.